util/util.go

```diff
@@ -307,7 +307,7 @@ func ClusterToInfrastructureMapFunc(gvk schema.GroupVersionKind) handler.ToReque
 // GetOwnerMachine returns the Machine object owning the current resource.
 func GetOwnerMachine(ctx context.Context, c client.Client, obj metav1.ObjectMeta) (*clusterv1.Machine, error) {
 	for _, ref := range obj.OwnerReferences {
-		if ref.Kind == "Machine" && ref.APIVersion == clusterv1.GroupVersion.String() {
+		if ref.Kind == "Machine" && strings.Split(ref.APIVersion, "/")[0] == clusterv1.GroupVersion.Group {
```

Comment:
We might want to consider using https://github.com/kubernetes/apimachinery/blob/76330795f827e44ffc47f4fdc8e0d39872bead8f/pkg/runtime/schema/group_version.go#L211 to parse the GroupVersion and compare the groups afterwards

Reply:
Thanks for the tip @vincepri 😄 

Changed to use `schema.ParseGroupVersion`